### PR TITLE
ieee802154: add config macros to config doc group

### PR DESCRIPTION
**Pull request by jia200x**
### Contribution description
See #10566 

I also added some `@brief` tags to these macros, and moved the `/* in dBm */` comment to the brief of `IEEE802154_DEFAULT_TXPOWER`

### Testing procedure
Compile documentation. A new group for the configuration macros of `net_ieee802154` should show up both in the `config` group doc as well in the doc of the respective `net_ieee802154` (sub-)module changed here.

### Issues/PRs references
Addresses #10566 in part

### CODE REVIEW

File: sys/include/net/ieee802154.h [hunk 1]
```diff
@@ -112,28 +112,43 @@ extern const uint8_t ieee802154_addr_bcast[IEEE802154_ADDR_BCAST_LEN];
 /** @} */
 
 /**
+ * @defgroup net_ieee802154_conf     Default values for IEEE802.15.4
```
Comment (miri64) [inline sys/include/net/ieee802154.h]: Unifying with the other configuration groups

```suggestion
 * @defgroup net_ieee802154_conf     IEEE802.15.4 compile configurations
```

?

Comment (jia200x) [reply sys/include/net/ieee802154.h]: roger

File: sys/include/net/ieee802154.h [hunk 2]
```diff
@@ -112,28 +112,43 @@ extern const uint8_t ieee802154_addr_bcast[IEEE802154_ADDR_BCAST_LEN];
 /** @} */
 
 /**
+ * @defgroup net_ieee802154_conf     IEEE802.15.4 compile configurations
```
Comment (miri64) [inline sys/include/net/ieee802154.h]: Very nit-picky, but the alignment is off by one now.

**Comment by miri64** [comment]
Other than that. `make doc` produces the expected output. You may squash immediately.

File: sys/include/net/ieee802154.h (hunk 2 above)
Comment (jia200x) [reply sys/include/net/ieee802154.h]: alignment with which block?

Comment (miri64) [reply sys/include/net/ieee802154.h]: The 4-tab spacing.

Comment (miri64) [reply sys/include/net/ieee802154.h]: Should be 
```suggestion
 * @defgroup net_ieee802154_conf    IEEE802.15.4 compile configurations
```

**Pull request merged**